test/specs/components/Button/Button-test.tsx

```diff
@@ -4,10 +4,15 @@ import { isConformant, handlesAccessibility } from 'test/specs/commonTests'
 import { getTestingRenderedComponent, mountWithProvider } from 'test/utils'
 
 import Button from 'src/components/Button/Button'
+import Icon from 'src/components/Icon/Icon'
+
 import { MenuBehavior } from 'src/lib/accessibility'
 
 describe('Button', () => {
-  isConformant(Button)
+  isConformant(Button).hasConformantShorthandProperty('icon', Icon, {
```

Comment:
I don't think we should couple common tests with a chaining syntax like this.  Inevitably, we are going to want to use a chained shorthand test without first passing a component to `isConformant`.  Would prefer simple and explicit imports and separate calls here.

Propose we stick with the original test name of `implementsShorthandProp`.  The base test is called `isConformant` as it tests many baseline standards ensuring the component conforms to them.  We don't need to call every test "conformant" when dealing with isolated cases such as "shorthand prop", "className", "accessibility", etc.  See previous prop vs property comment.

Reply:
let me, please, share my thoughts on that.

> I don't think we should couple common tests with a chaining syntax like this. Inevitably, we are going to want to use a chained shorthand test without first passing a component to isConformant

actually, it is very hard for me to imagine this situation. Our library is about introducing only conformant components (as this is the main point of it - to ensure this consistency between all the components). Could you, please, suggest some hypothetical case where it might happen?  Sorry, cannot come up with any by myself :(

Also, it seems that there are two aspects being mixed a bit. I am not against introducing it as a separate module, so that this functionality will be decoupled and modularized. At the same time, the question of the interface under which this functionality will be exposed to developer (chained calls, direct imports, etc) - this seems to be a separate question. The reason I might prefer chained syntax is that it is much more intuitive for the developer which testing methods she might consider (as well as that this chain, generally, describe the workflow for her):
- first, ensure that provided component is conformant
- then, optionally, ensure shorthand properties are properly handled
- ensure other aspects

And instead of looking on other specs for proper import statements, developer could rely on IntelliSense support:

![image](https://user-images.githubusercontent.com/9024564/44375329-65feb680-a4f3-11e8-970d-6ecbeaf79fa3.png)

Reply:
### Responses

Examples of non-conformant components are Portal and Provider.  Neither of these will pass the conformance test as they don't render typical UI components yet they still require other common tests.  The Provider needs `rendersChildren` while the Portal needs `hasSubcomponents` and likely `handlesAccessibility`.

The list of components that will not pass `isConformant` will grow.  In SUIR, we currently do not run conformance tests on `Ref`, `TransitionalPortal`, `MountNode`, or `Transition`.

Note that intellisense is still provided for all common tests when importing:

![image](https://user-images.githubusercontent.com/5067638/44381313-8bc09700-a4c4-11e8-9497-356fbe5b7fee.png)

### Proposal

Nonetheless, you raise a good point and perhaps we need to rethink what it means to be "conformant" if we expect many components to not conform.

In SUIR we were focused on making sure common UI components used by users conformed to many guidelines.  Overtime, some components fell outside of this scope or had difficulty abstracting good tests around them.  Stardust UI implemented conformance tests in largely the same way, but with some subtle differences and more capabilities given our `*.info.json` files for each component.

Maybe we trim down our conformance tests to strictly include tests that are applicable to _all_ components.  Then add each test that does not apply to all components as chained calls?  One upside is that all components can be handed to `isConformant`.

One downside to this is that it makes more room for error for most devs when creating most components (common UI components).  It is nice to simply say `isConformant` and fix errors.  However, having to know which chained calls they need to add could be daunting, especially if there are many.

Thoughts?

Reply:
agree with downsides of segregating tests, those will be especially critical now as we haven't critical mass of components for making a proper judgement. Lets follow the simplest path for now (the reason I wasn't sure about it before is that I wasn't aware of Provider and other components' specifics - thanks for the explanation):
- split `isConformant` and optional tests (such as `implementsShorthandProp `) to separate modules
- use direct module imports

We can reconsider this decision in future as we'll have more mature set of components (in case if necessary), so that it will be easier to make all the necessary analysis for these decisions.